doc/source/dev/index.rst

```diff
@@ -13,7 +13,7 @@ Contributing to NumPy
    development_environment
    development_workflow
    ../benchmarking
-   style_guide
+   NumPy C style guide <https://numpy.org/neps/nep-0045-c_style_guide.html>
```

Comment:
I wonder... Does `nep:NEP45` work?

Reply:
I just gave it a try and this (and several other iterations) didn't work. AFAICT `toctree` only supports using relative paths or external links, but does not parse roles (or role-like targets) for the entries.

Reply:
Oops, that's what I meant by "role-like target" (I'm not fluent with sphinx terminology). Using `NumPy C style guide <neps:NEP45>` or just `neps:NEP45` results in:
```
WARNING: toctree contains reference to nonexisting document 'dev/neps:NEP45'
```